source/Target/SwiftLanguageRuntime.cpp

```diff
@@ -3872,6 +3890,83 @@ SwiftLanguageRuntime::GetBridgedSyntheticChildProvider(ValueObject &valobj) {
   return nullptr;
 }
 
+void SwiftLanguageRuntime::WillStartExecutingUserExpression() {
+  std::lock_guard<std::mutex> lock(m_active_user_expr_mutex);
+  Log *log(GetLogIfAnyCategoriesSet(LIBLLDB_LOG_EXPRESSIONS));
+
+  if (m_active_user_expr_count == 0 &&
+      m_dynamic_exclusivity_flag_addr) {
+    // We're executing the first user expression. Toggle the flag.
+    Error error;
+    size_t bytes_read =
+      m_process->DoReadMemory(*m_dynamic_exclusivity_flag_addr,
+                              &m_original_dynamic_exclusivity_flag_state,
+                              sizeof(m_original_dynamic_exclusivity_flag_state),
+                              error);
```

Comment:
You are assuming that the size of a Host bool and the size of a Target bool are the same here.  That's not guaranteed to be true.  You can get the TypeSystem from the Target (GetScratchTypeSystemForLanguage is good, that's the expression context.) then get call GetBuiltinTypeByName("bool") then get the size from the type you got back.